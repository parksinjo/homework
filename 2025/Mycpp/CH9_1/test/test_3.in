1
2
14
P p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
S s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21: Computer 3.3 2
W w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul: Kia CEO
A a3 14 54 false :Dong-gu, Incheon: Physics 3.8 1 Kakao Manager :SK, LG, KAI: true
S s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan: Biology 3.8 3
W w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul: Naver Department-Head
A a4 17 88 true :Kangdong-gu, Daejeon: Electronics 3.4 2 NC Developer :CU, GS: false
P p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
S s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21: Computer 3.3 2
W w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul: Kia CEO
A a3 14 54 false :Dong-gu, Incheon: Physics 3.8 1 Kakao Manager :SK, LG, KAI: true
S s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan: Biology 3.8 3
W w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul: Naver Department-Head
A a4 17 88 true :Kangdong-gu, Daejeon: Electronics 3.4 2 NC Developer :CU, GS: false
0
0
