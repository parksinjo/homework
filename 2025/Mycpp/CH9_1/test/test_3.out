
******************************* Main Menu *********************************
* 0.Exit 1.PersonManager(ch3_2, 4, 6, 7_2, 8, 9)                          *
* 2.Class:Object(ch3_1) 3.CopyConstructor(ch5_1) 4.AllocatedMember(ch5_2) *
* 5.OperatorOverload(ch7_1) 6.Inheritance(ch8)                            *
***************************************************************************
Menu item number? 1
Person::Person(...):p0 10 70 false :Gwangju Nam-gu Bongseon-dong 21:
Person::Person(...):p1 11 61.1 true :Jong-ro 1-gil, Jongno-gu, Seoul:
Person::Person(...):p2 12 52.2 false :1001, Jungang-daero, Yeonje-gu, Busan:
Person::Person(...):p3 13 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
Person::Person(...):p4 14 64.4 false :88 Gongpyeong-ro, Jung-gu, Daegu:
Person::Person(...):s1 1 65.4 true :Jongno-gu Seoul:
Student::Student(...): Physics 3.8 1
Person::Person(...):s2 2 54.3 false :Yeonje-gu Busan:
Student::Student(...): Electronics 2.5 4
Person::Person(...):w1 3 33.3 false :Kangnam-gu Seoul:
Worker::Worker(...): Samsung Director
Person::Person(...):w2 4 44.4 true :Dobong-gu Kwangju:
Worker::Worker(...): Hyundai Manager
Person::Person(...):a1 5 55.5 true :Dong-gu Incheon:
Student::Student(...): Computer 3.5 2
Worker::Worker(...): Hyundai Labor
StudentWorker::StudentWorker(...): :CU KangNam,Seven Eleven,GSStore Suwon: false
Person::Person(...):a2 6 66.6 false :Sasang-gu Sejong:
Student::Student(...): History 3.1 1
Worker::Worker(...): Kia CEO
StudentWorker::StudentWorker(...): :Seven Eveven,eMart Jinju,CU Bongsun: true
Person::Person(const Person&):p0 10 70 false :Gwangju Nam-gu Bongseon-dong 21:
Person::Person(const Person&):p1 11 61.1 true :Jong-ro 1-gil, Jongno-gu, Seoul:
Person::Person(const Person&):s1 1 65.4 true :Jongno-gu Seoul:
Student::Student(const Student& s): Physics 3.8 1
Person::Person(const Person&):s2 2 54.3 false :Yeonje-gu Busan:
Student::Student(const Student& s): Electronics 2.5 4
Person::Person(const Person&):w1 3 33.3 false :Kangnam-gu Seoul:
Worker::Worker(const Worker& w): Samsung Director
Person::Person(const Person&):w2 4 44.4 true :Dobong-gu Kwangju:
Worker::Worker(const Worker& w): Hyundai Manager
Person::Person(const Person&):a1 5 55.5 true :Dong-gu Incheon:
Student::Student(const Student& s): Computer 3.5 2
Worker::Worker(const Worker& w): Hyundai Labor
StudentWorker::StudentWorker(const StudentWorker& a): :CU KangNam,Seven Eleven,GSStore Suwon: false
Person::Person(const Person&):a2 6 66.6 false :Sasang-gu Sejong:
Student::Student(const Student& s): History 3.1 1
Worker::Worker(const Worker& w): Kia CEO
StudentWorker::StudentWorker(const StudentWorker& a): :Seven Eveven,eMart Jinju,CU Bongsun: true
PersonManager::run() starts

====================== Person Management Menu ===================
= 0.Exit 1.Display 2.Append 3.Clear 4.Login(CurrentUser, ch9)   =
= 5.Insert(6_2) 6.Delete(6_2) 7.CopyPersons(7_3) 8.Reset(7_3)   =
=================================================================
Menu item number? 2
The number of persons to append? 14
Input 14 [delimiter(P, S, W, or A)] [person information] :
Person::Person(...): 0 0 false ::
P p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
S s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21: Computer 3.3 2
Person::Person(...): 0 0 false ::
Worker::Worker(...):  
W w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul: Kia CEO
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
Worker::Worker(...):  
StudentWorker::StudentWorker(...): :: false
A a3 14 54 false :Dong-gu, Incheon: Physics 3.8 1 Kakao Manager :SK, LG, KAI: true
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
S s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan: Biology 3.8 3
Person::Person(...): 0 0 false ::
Worker::Worker(...):  
W w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul: Naver Department-Head
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
Worker::Worker(...):  
StudentWorker::StudentWorker(...): :: false
A a4 17 88 true :Kangdong-gu, Daejeon: Electronics 3.4 2 NC Developer :CU, GS: false
Person::Person(...): 0 0 false ::
P p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
S s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21: Computer 3.3 2
Person::Person(...): 0 0 false ::
Worker::Worker(...):  
W w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul: Kia CEO
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
Worker::Worker(...):  
StudentWorker::StudentWorker(...): :: false
A a3 14 54 false :Dong-gu, Incheon: Physics 3.8 1 Kakao Manager :SK, LG, KAI: true
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
S s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan: Biology 3.8 3
Person::Person(...): 0 0 false ::
Worker::Worker(...):  
W w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul: Naver Department-Head
Person::Person(...): 0 0 false ::
Student::Student(...):  0 0
Worker::Worker(...):  
StudentWorker::StudentWorker(...): :: false
A a4 17 88 true :Kangdong-gu, Daejeon: Electronics 3.4 2 NC Developer :CU, GS: false
display(): count 22
[0] p0 10 70 false :Gwangju Nam-gu Bongseon-dong 21:
[1] p1 11 61.1 true :Jong-ro 1-gil, Jongno-gu, Seoul:
[2] s1 1 65.4 true :Jongno-gu Seoul: Physics 3.8 1
[3] s2 2 54.3 false :Yeonje-gu Busan: Electronics 2.5 4
[4] w1 3 33.3 false :Kangnam-gu Seoul: Samsung Director
[5] w2 4 44.4 true :Dobong-gu Kwangju: Hyundai Manager
[6] a1 5 55.5 true :Dong-gu Incheon: Computer 3.5 2 Hyundai Labor :CU KangNam,Seven Eleven,GSStore Suwon: false
[7] a2 6 66.6 false :Sasang-gu Sejong: History 3.1 1 Kia CEO :Seven Eveven,eMart Jinju,CU Bongsun: true
[8] p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
[9] s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21: Computer 3.3 2
[10] w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul: Kia CEO
[11] a3 14 54 false :Dong-gu, Incheon: Physics 3.8 1 Kakao Manager :SK, LG, KAI: true
[12] s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan: Biology 3.8 3
[13] w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul: Naver Department-Head
[14] a4 17 88 true :Kangdong-gu, Daejeon: Electronics 3.4 2 NC Developer :CU, GS: false
[15] p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
[16] s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21: Computer 3.3 2
[17] w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul: Kia CEO
[18] a3 14 54 false :Dong-gu, Incheon: Physics 3.8 1 Kakao Manager :SK, LG, KAI: true
[19] s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan: Biology 3.8 3
[20] w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul: Naver Department-Head
[21] a4 17 88 true :Kangdong-gu, Daejeon: Electronics 3.4 2 NC Developer :CU, GS: false

====================== Person Management Menu ===================
= 0.Exit 1.Display 2.Append 3.Clear 4.Login(CurrentUser, ch9)   =
= 5.Insert(6_2) 6.Delete(6_2) 7.CopyPersons(7_3) 8.Reset(7_3)   =
=================================================================
Menu item number? 0
PersonManager::run() returned
Person::~Person():p0 10 70 false :Gwangju Nam-gu Bongseon-dong 21:
Person::~Person():p1 11 61.1 true :Jong-ro 1-gil, Jongno-gu, Seoul:
Student::~Student(): Physics 3.8 1
Person::~Person():s1 1 65.4 true :Jongno-gu Seoul:
Student::~Student(): Electronics 2.5 4
Person::~Person():s2 2 54.3 false :Yeonje-gu Busan:
Worker::~Worker(): Samsung Director
Person::~Person():w1 3 33.3 false :Kangnam-gu Seoul:
Worker::~Worker(): Hyundai Manager
Person::~Person():w2 4 44.4 true :Dobong-gu Kwangju:
StudentWorker::~StudentWorker(): :CU KangNam,Seven Eleven,GSStore Suwon: false
Worker::~Worker(): Hyundai Labor
Student::~Student(): Computer 3.5 2
Person::~Person():a1 5 55.5 true :Dong-gu Incheon:
StudentWorker::~StudentWorker(): :Seven Eveven,eMart Jinju,CU Bongsun: true
Worker::~Worker(): Kia CEO
Student::~Student(): History 3.1 1
Person::~Person():a2 6 66.6 false :Sasang-gu Sejong:
Person::~Person():p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
Student::~Student(): Computer 3.3 2
Person::~Person():s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21:
Worker::~Worker(): Kia CEO
Person::~Person():w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul:
StudentWorker::~StudentWorker(): :SK, LG, KAI: true
Worker::~Worker(): Kakao Manager
Student::~Student(): Physics 3.8 1
Person::~Person():a3 14 54 false :Dong-gu, Incheon:
Student::~Student(): Biology 3.8 3
Person::~Person():s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan:
Worker::~Worker(): Naver Department-Head
Person::~Person():w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul:
StudentWorker::~StudentWorker(): :CU, GS: false
Worker::~Worker(): NC Developer
Student::~Student(): Electronics 3.4 2
Person::~Person():a4 17 88 true :Kangdong-gu, Daejeon:
Person::~Person():p3 11 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
Student::~Student(): Computer 3.3 2
Person::~Person():s3 12 71.5 false :Gwangju Nam-gu Bongseon-dong 21:
Worker::~Worker(): Kia CEO
Person::~Person():w3 13 65 true :Jong-ro 1-gil, Jongno-gu, Seoul:
StudentWorker::~StudentWorker(): :SK, LG, KAI: true
Worker::~Worker(): Kakao Manager
Student::~Student(): Physics 3.8 1
Person::~Person():a3 14 54 false :Dong-gu, Incheon:
Student::~Student(): Biology 3.8 3
Person::~Person():s4 15 80 true :1001, Jungang-daero, Yeonje-gu, Busan:
Worker::~Worker(): Naver Department-Head
Person::~Person():w4 16 77 false :Buk-ro 3, Kangdong-gu, Seoul:
StudentWorker::~StudentWorker(): :CU, GS: false
Worker::~Worker(): NC Developer
Student::~Student(): Electronics 3.4 2
Person::~Person():a4 17 88 true :Kangdong-gu, Daejeon:
StudentWorker::~StudentWorker(): :Seven Eveven,eMart Jinju,CU Bongsun: true
Worker::~Worker(): Kia CEO
Student::~Student(): History 3.1 1
Person::~Person():a2 6 66.6 false :Sasang-gu Sejong:
StudentWorker::~StudentWorker(): :CU KangNam,Seven Eleven,GSStore Suwon: false
Worker::~Worker(): Hyundai Labor
Student::~Student(): Computer 3.5 2
Person::~Person():a1 5 55.5 true :Dong-gu Incheon:
Worker::~Worker(): Hyundai Manager
Person::~Person():w2 4 44.4 true :Dobong-gu Kwangju:
Worker::~Worker(): Samsung Director
Person::~Person():w1 3 33.3 false :Kangnam-gu Seoul:
Student::~Student(): Electronics 2.5 4
Person::~Person():s2 2 54.3 false :Yeonje-gu Busan:
Student::~Student(): Physics 3.8 1
Person::~Person():s1 1 65.4 true :Jongno-gu Seoul:
Person::~Person():p4 14 64.4 false :88 Gongpyeong-ro, Jung-gu, Daegu:
Person::~Person():p3 13 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
Person::~Person():p2 12 52.2 false :1001, Jungang-daero, Yeonje-gu, Busan:
Person::~Person():p1 11 61.1 true :Jong-ro 1-gil, Jongno-gu, Seoul:
Person::~Person():p0 10 70 false :Gwangju Nam-gu Bongseon-dong 21:

******************************* Main Menu *********************************
* 0.Exit 1.PersonManager(ch3_2, 4, 6, 7_2, 8, 9)                          *
* 2.Class:Object(ch3_1) 3.CopyConstructor(ch5_1) 4.AllocatedMember(ch5_2) *
* 5.OperatorOverload(ch7_1) 6.Inheritance(ch8)                            *
***************************************************************************
Menu item number? 0
Good bye!!
